napari/_qt/widgets/qt_extension2reader.py

```diff
@@ -171,7 +173,7 @@ def _filter_compatible_readers(self, new_pattern):
         readers = self._npe2_readers.copy()
         to_delete = []
         compatible_readers = get_potential_readers(new_pattern)
-        for plugin_name, display_name in readers.items():
+        for plugin_name, _display_name in readers.items():
             if plugin_name not in compatible_readers:
                 to_delete.append(plugin_name)
```

Comment:
```suggestion
        for plugin_name in readers.keys(): 
            if plugin_name not in compatible_readers:
                to_delete.append(plugin_name)
```

Assuming `reader.keys()` exists.

Reply:
Fixed.